src/Text/XML/DSig.hs

```diff
@@ -202,11 +214,16 @@ verifyRoot creds el = do
 -- | Try a list of creds against a document.  If all fail, return a list of errors for each cert; if
 -- *any* succeed, return the empty list.
 verifyIO :: NonEmpty SignCreds -> LBS -> String -> IO [(SignCreds, Either HS.SignatureError ())]
-verifyIO creds el signedID = do
+verifyIO creds el signedID = capture' $ do
   results <- NL.zip creds <$> forM creds (\cred -> verifyIO' cred el signedID)
   case NL.filter (isRight . snd) results of
     (_:_) -> pure []
     []    -> pure $ NL.toList results
+  where
+    capture' :: IO a -> IO a
```

Comment:
It would be nicer if there was a comment explicitly complaining about libxml2 being noisy, but ok.

Reply:
4c6e74fdef63db24e66fbe88656afe8833c72ab8